File: JScript/clsPfCoordinate.js
//
// Copyright (c)2014 S C Harrison
// Refer to License.txt for terms and conditions of use.
//

//Define Class
function clsPfCoordinate()
{
  this.key = 0;
  this.x = 0;           //.. x-coord of a joint ..
  this.y = 0;           //.. y-coord of a joint ..
}


clsPfCoordinate.prototype.initialise = function()
{
  this.key = 0;
  this.x = 0;
  this.y = 0;
}

clsPfCoordinate.prototype.setValues = function(nodeKey,x1,y1)
{
    this.key = nodeKey;
    this.x = x1;
    this.y = y1;
}

clsPfCoordinate.prototype.sprint = function()
{
  var s;
  
  s = StrLPad(this.key.toString(), 8) + StrLPad(this.x.toString(), 12) + StrLPad(this.y.toString(), 12);
  
  return s
}

clsPfCoordinate.prototype.cprint = function()
{
  WScript.Echo(this.sprint());
}

clsPfCoordinate.prototype.fprint = function(fp)
{
  fp.WriteLine(this.sprint());
}

clsPfCoordinate.prototype.fgetData = function(fp)
{
  var s;
  var n;
  var dataflds;
  var i,n;

  WScript.Echo( "fgetData ...");
  
   s = fp.ReadLine();
   //WScript.Echo( s);
   this.sgetData(s);
   
   WScript.Echo( "... fgetData");
}


clsPfCoordinate.prototype.sgetData = function(s)
{
  var n;
  var dataflds;
  var i,n;

  WScript.Echo( "sgetData ...");

  //WScript.Echo( s);

   s = s.replace(/^\s+|\s+$/gm,''); //trim trailing spaces
   dataflds = s.match(/-?\d+(?:[,.]\d+)?/gi);


    this.key = parseInt(dataflds[0]);
    this.x = parseFloat(dataflds[1]);
    this.y = parseFloat(dataflds[2]);
   
   WScript.Echo( "... sgetData");
}

File: VisualStudio/CSharp/w7cpframe/clsPfCoordinate.cs
//
// Copyright (c)2016 S C Harrison
// Refer to License.txt for terms and conditions of use.
//
using System;
using System.Collections.Generic;
using System.Text;
using System.IO;
using System.Text.RegularExpressions;

namespace w7cpframe
{
    class clsPfCoordinate
    {
        //Define Class

        public int key = 0;
        public double x = 0;           //.. x-coord of a joint ..
        public double y = 0;           //.. y-coord of a joint ..



        public void initialise()
        {
            key = 0;
            x = 0;
            y = 0;
        }

        public void setValues(int nodeKey, double x1, double y1)
        {
            key = nodeKey;
            x = x1;
            y = y1;
        }

        public string sprint()
        {
            string s;

            s = key.ToString().PadLeft(8) + x.ToString().PadLeft(12) + y.ToString().PadLeft(12);

            return s;
        }

        public void cprint()
        {
            System.Console.WriteLine(sprint());
        }

        public void fprint(StreamWriter fp)
        {
            fp.WriteLine(sprint());
        }

        public void fgetData(StreamReader fp)
        {
            string s;

            System.Console.WriteLine("fgetData ...");

            s = fp.ReadLine();
            sgetData(s);

            System.Console.WriteLine("... fgetData");
        }


        public void sgetData(string s)
        {
            string[] dataflds = new string[10]; //(0 To 9);
            char[] delimters = new char[] { ' ', '\t' };
            int i;


            System.Console.WriteLine("sgetData ...");

            string regResult;
            string pattern = @"^\s+|\s+$"; //trim trailing spaces
            regResult = Regex.Replace(s, pattern, "");

            dataflds.Initialize();
            string pattern2 = @"-?\d+(?:[,.]\d+)?";
            i = 0;
            foreach (Match match in Regex.Matches(regResult, pattern2))
            {
                if (match.Value != "")
                {
                    dataflds[i] = match.Value;
                    System.Console.WriteLine(match.Value);
                    i++;
                }
            }


            key = int.Parse(dataflds[0]);
            x = double.Parse(dataflds[1]);
            y = double.Parse(dataflds[2]);

            System.Console.WriteLine("... sgetData");
        }

    } // class
} //name space
